lib/identity_doc_auth/error_generator.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: true
+require 'identity_doc_auth/errors'
+
+module IdentityDocAuth
+  class UnknownDocAuthError < StandardError; end
+  class UnknownDocAuthAlert < StandardError; end
+
+  class ErrorGenerator
+    attr_reader :config
+
+    def initialize(config)
+      @config = config
+    end
+
+    # These constants are the key names for the TrueID errors hash that is returned
+    ID = :id
+    FRONT = :front
+    BACK = :back
+    SELFIE = :selfie
+    GENERAL = :general
+
+    ERROR_KEYS = [
+      ID,
+      FRONT,
+      BACK,
+      SELFIE,
+      GENERAL,
+    ].to_set.freeze
+
+    ALERT_MESSAGES = {
+      '1D Control Number Valid': { type: BACK, msg_key: Errors::REF_CONTROL_NUMBER_CHECK },
+      '2D Barcode Content': { type: BACK, msg_key: Errors::BARCODE_CONTENT_CHECK },
+      '2D Barcode Read': { type: BACK, msg_key: Errors::BARCODE_READ_CHECK },
+      'Birth Date Crosscheck': { type: ID, msg_key: Errors::BIRTH_DATE_CHECKS },
+      'Birth Date Valid': { type: ID, msg_key: Errors::BIRTH_DATE_CHECKS },
+      'Control Number Crosscheck': { type: BACK, msg_key: Errors::CONTROL_NUMBER_CHECK },
+      'Document Classification': { type: ID, msg_key: Errors::ID_NOT_RECOGNIZED },
+      'Document Crosscheck Aggregation': { type: ID, msg_key: Errors::DOC_CROSSCHECK },
+      'Document Expired': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Document Number Crosscheck': { type: ID, msg_key: Errors::DOC_NUMBER_CHECKS },
+      'Expiration Date Crosscheck': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Expiration Date Valid': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Full Name Crosscheck': { type: ID, msg_key: Errors::FULL_NAME_CHECK },
+      'Issue Date Crosscheck': { type: ID, msg_key: Errors::ISSUE_DATE_CHECKS },
+      'Issue Date Valid': { type: ID, msg_key: Errors::ISSUE_DATE_CHECKS },
+      'Layout Valid': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Near-Infrared Response': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Sex Crosscheck': { type: ID, msg_key: Errors::SEX_CHECK },
+      'Visible Color Response': { type: ID, msg_key: Errors::VISIBLE_COLOR_CHECK },
+      'Visible Pattern': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Visible Photo Characteristics': { type: FRONT, msg_key: Errors::VISIBLE_PHOTO_CHECK },
+    }.freeze
+
+    # rubocop:disable Metrics/PerceivedComplexity
+    def generate_doc_auth_errors(response_info)
+      liveness_enabled = response_info[:liveness_enabled]
+      alert_error_count = response_info[:alert_failure_count]
+
+      unknown_fail_count = scan_for_unknown_alerts(response_info)
+      alert_error_count -= unknown_fail_count
+
+      im_errors = get_image_metric_errors(response_info[:image_metrics])
+      return im_errors unless im_errors.empty?
```

Comment:
do we enough space to expand `im`?

```suggestion
      image_metric_errors = get_image_metric_errors(response_info[:image_metrics])
      return image_metric_errors unless image_metric_errors.empty?
```

Reply:
yeah, I'll expand it.